From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Wed, 21 Jan 2026 22:01:49 +0100
Subject: guniprop: Do not convert size_t to gint

We were correctly using size_t in output_special_case() since commit
362f92b69, but then we converted the value back to int

Related to: #3872

ID: #YWH-PGM9867-171
Bug: https://gitlab.gnome.org/GNOME/glib/-/issues/3872
Bug-CVE: CVE-2026-1489
Bug-Debian: https://bugs.debian.org/1126549
Applied-upstream: 2.87.3, commit:a27e7bdd9e5a37bd188925060931a23427ebb92c
---
 glib/guniprop.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/glib/guniprop.c b/glib/guniprop.c
index 0ee9aa3..ef4d6bc 100644
--- a/glib/guniprop.c
+++ b/glib/guniprop.c
@@ -798,7 +798,7 @@ output_marks (const char **p_inout,
   return len;
 }
 
-static gint
+static size_t
 output_special_case (gchar *out_buffer,
 		     int    offset,
 		     int    type,
